ray-operator/controllers/ray/batchscheduler/scheduler-plugins/scheduler_plugins_test.go

```diff
@@ -0,0 +1,94 @@
+package schedulerplugins
+
+import (
+	"testing"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+)
+
+func createTestRayCluster(numOfHosts int32) rayv1.RayCluster {
+	headSpec := corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name: "ray-head",
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("500m"),
+						corev1.ResourceMemory: resource.MustParse("512Mi"),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("256m"),
+						corev1.ResourceMemory: resource.MustParse("256Mi"),
+					},
+				},
+			},
+		},
+	}
+
+	workerSpec := corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name: "ray-worker",
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("500m"),
+						corev1.ResourceMemory: resource.MustParse("512Mi"),
+						"nvidia.com/gpu":      resource.MustParse("1"),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("256m"),
+						corev1.ResourceMemory: resource.MustParse("256Mi"),
+					},
+				},
+			},
+		},
+	}
+
+	return rayv1.RayCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "raycluster-sample",
+			Namespace: "default",
+		},
+		Spec: rayv1.RayClusterSpec{
+			HeadGroupSpec: rayv1.HeadGroupSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: headSpec,
+				},
+			},
+			WorkerGroupSpecs: []rayv1.WorkerGroupSpec{
+				{
+					Template: corev1.PodTemplateSpec{
+						Spec: workerSpec,
+					},
+					Replicas:    ptr.To[int32](2),
+					NumOfHosts:  numOfHosts,
+					MinReplicas: ptr.To[int32](1),
+					MaxReplicas: ptr.To[int32](4),
+				},
+			},
+		},
+	}
+}
+
+func TestCalculateDesiredResources(t *testing.T) {
+	a := assert.New(t)
+
+	cluster := createTestRayCluster(1)
+
+	totalResource := utils.CalculateDesiredResources(&cluster)
+
+	// 256m * 3 (requests, not limits)
+	a.Equal("768m", totalResource.Cpu().String())
+
+	// 256Mi * 3 (requests, not limits)
+	a.Equal("768Mi", totalResource.Memory().String())
+
+	// 2 GPUs total
+	a.Equal("2", totalResource.Name("nvidia.com/gpu", resource.BinarySI).String())
+}
```

Comment:
What are you testing here? It seems that you didn't test anything specific to the kube scheduler plugin.

Reply:
Currently we only calculate the min-resource and create the podgroup. The functionality of creating the podgroup is completed by the client. So in unit test we just need to test the logic of calculate the min-resource.

Reply:
Sure